### ISSUE TITLE
Provisioning API returns generic error for invalid duration

### ISSUE THREAD

**Issue by grobinson-grafana** [grafana/grafana#56387]
**What happened**:

The Provisioning API returns the following generic error when the `For` field is invalid:

```
{"message":"bad request data","traceID":""}
```

The error is logged here:

```
ERROR[10-05|11:21:33] bad request data logger=context userId=1 orgId=1 uname=admin error="not a valid duration string: \"300\"" remote_addr=127.0.0.1 traceID=
```

However it should also be sent in the response.

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Grafana version:
- Data source type & version:
- OS Grafana is installed on:
- User OS & Browser:
- Grafana plugins:
- Others:

**Comment by vickymhs** [comment]
Hi, I would like to work on this. It would be great if you can provide some resources on the API and how to get started with this. Looking forward to learn and contribute

**Comment by neel1996** [comment]
Hi @grobinson-grafana, 

I have checked the `/api/v1/provisioning/alert-rules` endpoint and received the expected error in the response when the `For` field is not a valid duration.

**Request:**

```json
{
    "Condition": "A",
    "Data": [
        {
            "datasourceUid": "-100",
            "model": {
                "conditions": [
                    {
                        "evaluator": {
                            "params": [
                                0,
                                0
                            ],
                            "type": "gt"
                        },
                        "operator": {
                            "type": "and"
                        },
                        "query": {
                            "params": []
                        },
                        "reducer": {
                            "params": [],
                            "type": "avg"
                        },
                        "type": "query"
                    }
                ],
                "datasource": {
                    "type": "__expr__",
                    "uid": "__expr__"
                },
                "expression": "1 == 1",
                "hide": false,
                "intervalMs": 1000,
                "maxDataPoints": 43200,
                "refId": "A",
                "type": "math"
            },
            "queryType": "",
            "refId": "A",
            "relativeTimeRange": {
                "from": 0,
                "to": 0
            }
        }
    ],
    "ExecErrState": "OK",
    "FolderUID": "project_x",
    "NoDataState": "OK",
    "OrgID": 1,
    "RuleGroup": "eval_group_1",
    "Title": "Always firing",
    "for": "300"
}
```

**Response:**

```json
{
    "error": "not a valid duration string: \"300\"",
    "message": "bad request data",
    "traceID": ""
}
```

**Error log:**
```
ERROR[10-08|16:50:34] bad request data                         logger=context userId=1 orgId=1 uname=admin error="not a valid duration string: \"300\"" remote_addr=[::1] traceID=
```

Could you please clarify if this issue is addressing a different API or the same one mentioned above.

**Comment by A-5ingh** [comment]
This is not reproducible on either the latest commit or `1bcce79`. @grobinson-grafana I hope you are running it locally on using `development` mode. The only way you won't see error if you using `prod` settings.

**Comment by grobinson-grafana** [comment]
Hi! :wave: This issue is about returning information about errors in the client request (i.e. invalid data) even when Grafana is run in production mode, as otherwise customers that use the Provisioning API in Hosted Grafana will find it hard to figure out what is wrong with their request.

**Comment by davidifr** [comment]
Hi @grobinson-grafana ,
I would like to pick this up.
Just a small question so I would better understand the solution here.

You want to modify all of the "bad request data" messages in the provision API to show the error which is basically the err.Error() string right?

Also, you wouldn't mind that in `development` mode you see the same error string on both message and error fields?
`{message: "not a valid duration string: \"300\"", error: "not a valid duration string: \"300\""}`
Again, cause of the above mentioned [condition](https://github.com/grafana/grafana/blob/53baecd71f5519e0ef300f3067cb89eab2908e2f/pkg/api/response/response.go#L213).

**Comment by armandgrillet** [comment]
Thanks @davidifr, we will review this.

**Comment by filip-aipl** [comment]
Any updates on this error message? I get this calling alert rules endpoint or dashboard endpoint.

**Comment by domdfcoding** [comment]
Ran into this myself. It does seem rather daft to show the errors only when running in development mode. I'm not developing Grafana, I'm trying to use it, hence it's in production mode.

**Comment by CodeGovindz** [comment]
could you please assign this issue to me

**Comment by gkech** [comment]
Really interested in the resolution of this as well, 👍🏽 for prioritizing of this

**Comment by domdfcoding** [comment]
I don't think this has been fixed. If it hasn't it certainly should be (and #70785 resurrected. It was waiting for a review when the bot closed it)

**Comment by jgfranco17** [comment]
Hello, is this issue still open? If so, I would like to give it a try 😄

**Comment by davidwebdeveloper** [comment]
I've submitted a fix for this in #123995. The change updates the code generation template and all generated API handler files to pass `err.Error()` as the response message instead of the generic `"bad request data"` string. This way users will see the actual validation error (e.g., `"not a valid duration string: \"300\""`) in the API response.

**Comment by yuminn-k** [comment]
I opened a draft PR to address this: https://github.com/grafana/grafana/pull/127213

The PR updates the alerting API body binding error path so invalid request bodies can return the underlying validation error instead of the generic `bad request data` message. I also added a regression test for an invalid duration value.
